frontend/packages/dev-console/src/components/topology/PodRingDataController.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import { Firehose, FirehoseResource } from '@console/internal/components/utils';
+import { knativeServingResources } from '@console/knative-plugin/src/utils/create-knative-utils';
+import TransformPodRingData from './podring-utils';
+import { Resource } from './topology-types';
+
+export interface PodRingDataResources {
+  replicationControllers: Resource;
+  pods: Resource;
+  deploymentConfigs: Resource;
+  services: Resource;
+  deployments: Resource;
+  replicasets: Resource;
+  daemonSets: Resource;
+}
+
+export interface ControllerProps {
+  loaded?: boolean;
+  loadError?: any;
+  resources?: PodRingDataResources;
+  render(RenderProps): React.ReactElement;
+}
+
+export interface PodRingDataControllerProps {
+  namespace: string;
+  render(RenderProps): React.ReactElement;
+  knative: boolean;
+}
+
+const Controller: React.FC<ControllerProps> = React.memo(
+  ({ resources, render, loaded, loadError }) => {
+    return render({
+      loaded,
+      loadError,
+      data: loaded
+        ? new TransformPodRingData(resources)
+            .transformDataBy('deployments')
+            .transformDataBy('deploymentConfigs')
+            .transformDataBy('daemonSets')
+            .getPodsData()
+        : null,
+    });
+  },
+);
+
+const PodRingDataContoller: React.FC<PodRingDataControllerProps> = ({
+  namespace,
+  render,
+  knative,
+}) => {
+  let resources: FirehoseResource[] = [
```

Comment:
Seems like a lot of over fetching of data simply to display the pod status. eg. Why fetch DaemonSets while the main resource is a DeploymentConfig?